06/ht/src/mapService.js

```diff
@@ -0,0 +1,94 @@
+import { setSearchCoords, getCityData, getMarkerAndMoveLocation } from "./geolocationService";
+import { displayLastCities, displayLastFavorites } from "./pagesInfo";
+import { saveLocalStorage } from "./storageData";
+
+const favoritesUser = JSON.parse(localStorage.getItem("FavoritesUser")) || [];
+const dataUser = JSON.parse(localStorage.getItem("DataUser")) || { city: [] };
```

Comment:
- Тут ведь можно не хранить объект `city` а сразу массив хранить. И название не самое хорошее ключа. В моем понимании `history: []` or `cities: []` вот так нужно было хранить, без дополнительных вложенностей. 
```suggestion
const dataUser = JSON.parse(localStorage.getItem("history")) || [];
```

- Также у тебя есть файл storageData почему ты напрямую с localstorage работаешь, а не через этот файл? В общем у тебя должны быть вызовы только оттуда, а не работа с localstorage напрямую.